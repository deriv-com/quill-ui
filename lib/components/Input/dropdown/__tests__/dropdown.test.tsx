import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import InputDropdown, { DropdownOptionProps } from "../index";

// Mock options for dropdown
const options = [
    { value: 1, label: "Option 1" },
    { value: 2, label: "Option 2" },
    { value: 3, label: "Option 3" },
];

describe("InputDropdown component", () => {
    it("renders dropdown with options", () => {
        const dropdownProps: DropdownOptionProps = {
            options: options,
        };
        render(<InputDropdown {...dropdownProps} />);
        fireEvent.click(screen.getByTestId("dropdown-input"));

        options.forEach((option) => {
            expect(screen.getByText(option.label)).toBeInTheDocument();
        });
    });

    it("selects an option from dropdown", () => {
        const dropdownProps: DropdownOptionProps = {
            options: options,
        };
        render(<InputDropdown {...dropdownProps} />);

        fireEvent.click(screen.getByTestId("dropdown-input"));

        fireEvent.click(screen.getByText(options[0].label));

        expect(screen.getByTestId("dropdown-input")).toHaveValue(
            options[0].label,
        );
    });

    it("closes dropdown on selection", () => {
        const dropdownProps: DropdownOptionProps = {
            options: options,
        };
        render(<InputDropdown {...dropdownProps} />);

        fireEvent.click(screen.getByTestId("dropdown-input"));

        fireEvent.click(screen.getByText(options[0].label));

        expect(screen.queryByText(options[0].label)).not.toBeInTheDocument();
    });
    it("should render a TextField with success status", () => {
        const dropdownProps: DropdownOptionProps = {
            options: options,
        };
        const { container } = render(
            <InputDropdown {...dropdownProps} status="success" />,
        );
        expect(container).toMatchSnapshot();
    });

    it("should render a TextField with error status", () => {
        const dropdownProps: DropdownOptionProps = {
            options: options,
        };
        const { container } = render(
            <InputDropdown {...dropdownProps} status="error" />,
        );
        expect(container).toMatchSnapshot();
    });
});
