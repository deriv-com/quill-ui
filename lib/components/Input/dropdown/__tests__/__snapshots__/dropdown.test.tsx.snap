// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDropdown component should render a TextField with error status 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__size--md quill-input__wrapper__status--error"
    >
      <div
        class="quill-input-no-label__wrapper"
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          class="input peer input__align--left input__size--md"
          data-headlessui-state=""
          data-testid="dropdown-input"
          readonly=""
          role="combobox"
          type="select"
          value=""
        />
      </div>
      <svg
        class="dropdown__transform"
        data-state="close"
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="m6.535 14.715-5.25-5.25c-.273-.246-.273-.656 0-.93.246-.246.656-.246.93 0L7 13.348l4.785-4.786c.246-.273.656-.273.93 0a.644.644 0 0 1 0 .903l-5.277 5.25a.6.6 0 0 1-.903 0"
          />
        </g>
        <defs>
          <clippath
            id="f43687b6c6d919677eabf8a2c671e282__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
  </div>
</div>
`;

exports[`InputDropdown component should render a TextField with success status 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__size--md quill-input__wrapper__status--success"
    >
      <div
        class="quill-input-no-label__wrapper"
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          class="input peer input__align--left input__size--md"
          data-headlessui-state=""
          data-testid="dropdown-input"
          readonly=""
          role="combobox"
          type="select"
          value=""
        />
      </div>
      <svg
        class="dropdown__transform"
        data-state="close"
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="m6.535 14.715-5.25-5.25c-.273-.246-.273-.656 0-.93.246-.246.656-.246.93 0L7 13.348l4.785-4.786c.246-.273.656-.273.93 0a.644.644 0 0 1 0 .903l-5.277 5.25a.6.6 0 0 1-.903 0"
          />
        </g>
        <defs>
          <clippath
            id="f43687b6c6d919677eabf8a2c671e282__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
  </div>
</div>
`;
